src/modules/inventory/CMakeLists.txt

```diff
@@ -29,14 +29,14 @@ target_include_directories(Inventory PUBLIC
     ${COMMON_FOLDER}/data_provider/include
     ${COMMON_FOLDER}/dbsync/include
     ${COMMON_FOLDER}/error_messages/include
-    ${COMMON_FOLDER}/file_op/include
     ${COMMON_FOLDER}/hashHelper/include
     ${COMMON_FOLDER}/privsep_op/include
     ${COMMON_FOLDER}/pthreads_op/include
     ${COMMON_FOLDER}/stringHelper/include
     ${COMMON_FOLDER}/time_op/include
     ${COMMON_FOLDER}/timeHelper/include
-    ${COMMON_FOLDER}/utils)
+    ${COMMON_FOLDER}/utils
+    ${COMMON_FOLDER}/pal/include)
```

Comment:
If it is not, then it doesn't need to.
Several incompatibilities are just saved by using defines in the headers, not in the libraries (ex. popen & pclose, strtok_r, some required types that only are missed in MSVC, etc).

In that case, no linkage to libraries are necessary, just including pal.h (which will eventually include the header corresponding to the platform) is enough.

Reply:
I see your point. From my perspective, the path to a project's include folder feels like a 'lower-level' concern compared to the target itself, as the target inherently knows which folders to include publicly. Manually selecting include folders from different subprojects seems like a detail we could avoid by linking directly against the target.

If we don't need the symbol definitions, interface targets could serve the purpose effectively.

I wouldn’t say this approach is outright wrong, especially since it’s so common in our codebase, but for new code, I tend to prefer avoiding it in favor of a more target-centric approach.

Reply:
Done